src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,357 @@
+- **PlaceholderText**: input elements with a `placeholder`-attribute
+- **Text**: searches the text in all text nodes within the element, even if split over multiple nodes
+- **DisplayValue**: form elements showing the given value
+- **AltText**: images with the alt text
+- **Title**: HTML elements with the `title`-attribute or SVGs with the `<title>`-tag containing the given text
+- **TestId**: queries by the `data-testid`-attribute. You can setup a different attribute via `configure({testIdAttribute: 'data-my-test-attribute'})`
+
+For more information, check the [testing-library documentation](https://testing-library.com/docs/queries/about).
+
+#### Testing in context
+
+The component might be reliant on some context. In order to wrap it, use the `wrapper`-option. 
+
+```tsx title="Context.test.tsx"
+import { test, expect } from "vitest"
+import { render } from "@solidjs/testing-library"
+import { DataContext, DataConsumer } from "./Data"
+
+const wrapper = (props) => <DataContext value="test" {...props} />
+
+test("receives data from context", () => {
+  const { queryByText } = render(() => <DataConsumer />, { wrapper })
+  expect(getByText("test")).toBeInTheDocument()
+});
+```
+
+Wrappers can be re-used if they are created externally. For wrappers with different values, a higher-order component creating the required wrappers can make the tests more concise:
+
+```tsx
+const createWrapper = (value) => (props) => 
+  <DataContext value={value} {...props}/>
+```
+
+<Callout title="Using multiple providers">
+If using multiple providers, [solid-primitives](https://primitives.solidjs.community/) has [`<MultiProvider>`](https://primitives.solidjs.community/package/context#multiprovider) to avoid nesting multiple levels of providers
+</Callout>
+
+#### Testing routes
+
+For convenience, `render` supports the `location`-option that wraps the rendered component in a router pointing at the given location. Since the `<Router>` component is lazily loaded, the first query after rendering needs to be asynchronous, i.e. `findBy...`:
+
+```tsx
+const { findByText } = render(
+  () => <Route path="/article/:id" component={Article} />,
+  { location: "/article/12345" }
+);
+expect(await findByText("Article 12345")).toBeInTheDocument()
+```
+
+### Interacting with components
+
+Many components are not static, but instead change on user interactions. These interactions need to be simulated to test the changes. To simulate user interactions, the creators of testing-library have provided the `@testing-library/user-event` library. It takes care of the usual order of events as they would occur in actual user interactions, e.g. that a `click` is accompanied by `mousemove`, `hover`, `keydown`, `focus`, `keyup`, and `keypress`.
+
+If you want to use fake timers with `vi.useFakeTimers()` in your tests, be sure to set it up:
+
+```tsx frame="none"
+import { vi } from "vitest"
+
+const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime })
+```
```

Comment:
does it make sense to add the fakeTimers part here? If the focus of this section is interaction we should just mention userEvent normally. Perhaps the fakeTimers part can be an aside thing

Reply:
Let me think a bit on this one.

Reply:
I think I can combine it with the example.